types/src/execution_result.rs

```diff
@@ -185,12 +185,6 @@ impl ExecutionResult {
         &*EXECUTION_RESULT
     }
 
-    fn tag_byte(&self) -> u8 {
```

Comment:
Alternatively, this method could simply start returning `Option<u8>` and on callsite:
```rust
tag_byte().ok_or_else(|| bytesrepr::Error::Formatting)?;
```

Reply:
Sure. I changed it in all 3 similar places. Do you want me to push?

Reply:
I don't have preference to be honest, I actually liked the previous version more since it hide the conversion and could possibly be extended with other error conditions. But it isn't a big deal to be fair.

It's up to you and @Fraser999 .

Reply:
I didn't see the point in retaining the one-line private method which is only called in one place :)